tests/networking-hosts.nix

```
@@ -0,0 +1,27 @@
+{ config, pkgs, ... }:
+
+{
+  networking.hosts = {
+    "127.0.0.1" = [ "my.super.host" ];
+    "10.0.0.1" = [ "my.super.host" "my.other.host" ];
+  };
+
+  test = ''
+    echo checking /etc/hosts file >&2
+
+    file=${config.out}/etc/hosts
+
+    if ! grep '127.0.0.1' $file | head -n1 | grep localhost$; then
+      exit 1
+    fi
```

```suggestion
    grep '127.0.0.1' $file | head -n1 | grep localhost$
```

The `if !; then exit` isn't necessary